### ISSUE TITLE
[FEA][JSON reader] to support parsing with single quotes

### ISSUE THREAD

**Issue by wbo4958** [rapidsai/cudf#10004]
This is part of FEA of https://github.com/NVIDIA/spark-rapids/issues/9
We have a JSON file

``` json
{'name': 'Reynold Xin'}
```

CUDF can't parse this file because of 
``` console
ai.rapids.cudf.CudfException: cuDF failure at: /home/bobwang/work.d/nvspark/cudf/cpp/src/io/json/reader_impl.cu:609: Error determining column names.
```
Spark can parse it by default.

We expect there is a configure `allowSingleQuotes` to control this behavior.

**Comment by revans2** [comment]
By default Spark allows single quotes by default so this is a blocker for us to enable JSON parsing in Spark by default.

**Comment by GregoryKimball** [comment]
After some discussion it seems we could support this by adding new states for single-quoted field names and single-quoted string values. We would have to increase the size of the state machine which would bring a performance penalty. I'll leave this in backlog for now.

**Comment by revans2** [comment]
I think this is done now. @GregoryKimball @andygrove do you both agree?

**Comment by GregoryKimball** [comment]
Closed by #14729. Thank you @shrshi!
